packages/react-devtools-shared/src/devtools/views/Components/InspectedElementContext.js

```diff
@@ -78,54 +74,44 @@ export function InspectedElementContextController({children}: Props) {
   if (elementHasChanged) {
     setState({
       element,
-      inspectedPaths: {},
+      path: null,
     });
   }
 
   // Don't load a stale element from the backend; it wastes bridge bandwidth.
-  const inspectedElement =
-    !elementHasChanged && element !== null
-      ? inspectElement(
-          element,
-          state.inspectedPaths,
-          forceUpdateRef.current,
-          store,
-          bridge,
-        )
-      : null;
+  let inspectedElement = null;
+  if (!elementHasChanged && element !== null) {
+    inspectedElement = inspectElement(element, state.path, store, bridge);
+  }
 
   const inspectPaths: InspectPathFunction = useCallback<InspectPathFunction>(
     (path: Path) => {
       startTransition(() => {
-        forceUpdateRef.current = true;
-        setState(prevState => {
-          const cloned = {...prevState};
-          let current = cloned.inspectedPaths;
-          path.forEach(key => {
-            if (!current[key]) {
-              current[key] = {};
-            }
-            current = current[key];
-          });
-          return cloned;
+        setState({
+          element: state.element,
+          path,
         });
         refresh();
       });
     },
-    [setState],
+    [setState, state],
   );
 
-  // Force backend update when inspected paths change.
+  // Reset path
   useEffect(() => {
```

Comment:
Can you explain what this is for?

Reply:
Yeah, this is a little janky.

I'm temporarily storing the most recently-inspected (hydrated) path in state so when we get a new one, and there's a transition, the component asks the backend for it.

I'm not sure of a better way to do this yet. I guess I could have a totally separate suspense cache that's just dehydrated paths. Wouldn't want to tear though if a path was inspected after an update, but before the polled update check.

There may be a more graceful design here though. At least this commit restores the optimizations that I removed previously in the meanwhile.